tauri/src/alchemy/mod.rs

```diff
@@ -0,0 +1,87 @@
+mod error;
+mod global;
+
+pub use self::error::{Error, Result};
+use crate::{
+    db::DB,
+    networks::Networks,
+    settings::Settings,
+    types::GlobalState,
+    wallets::{WalletControl, Wallets},
+};
+use ethers_core::types::{Address, U256};
+use once_cell::sync::Lazy;
+use serde::Deserialize;
+use serde_json::json;
+
+use url::Url;
+
+use std::collections::HashMap;
+
+static ENDPOINTS: Lazy<HashMap<u32, Url>> = Lazy::new(|| {
+    let mut endpoints: HashMap<u32, Url> = HashMap::new();
+    endpoints.insert(
+        1,
+        Url::parse("https://eth-mainnet.g.alchemy.com/v2/").unwrap(),
+    );
+    endpoints.insert(
+        5,
+        Url::parse("https://eth-goerli.g.alchemy.com/v2/").unwrap(),
+    );
+    endpoints
+});
+
+#[derive(Debug)]
+pub struct Alchemy {
+    db: DB,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AlchemyResponse {
+    pub address: Address,
+    pub token_balances: Vec<TokenBalance>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TokenBalance {
+    pub contract_address: Address,
+    pub token_balance: U256,
+}
+
+impl Alchemy {
+    pub fn new(db: DB) -> Self {
+        Self { db }
+    }
+
+    async fn fetch_balances(&self) -> Result<()> {
```

Comment:
this should get the chain_id and address as parameter instead of fetching the current one

the idea is to decouple those concepts. Alchemy doesn't need to care what the "current" address is, since we may decide to fetch others in the background